ellipticoind/src/api/mutations.rs

```diff
@@ -44,3 +50,101 @@ impl Mutations {
+        Some(x) => return match x {
+            Burned(proof) => Rejected(bytes_from_signed_txs(proof.values())),
+            Accepted(witnessed_block) => Witnessed(bytes_from_signed_tx(witnessed_block)),
+        },
+        None => ()
+    }
+
+    if next_block.is_miner_burned_by_me(&next_block.miner).await {
+        return Rejected(bytes_from_signed_txs(next_block.current_burn_proof().unwrap().values()))
+    }
+
+    let block_u8: &[u8] = &(Vec::from(posted_block));
+    let witnessed_block: WitnessedMinerBlock = next_block.witness_miner_block(block_u8);
+
+    // TODO: Save witness here
+
+    let miner = block.apply(txs).await;
+    BLOCK_CHANNEL.0.send(miner.clone()).await;
+
+    next_block.increment(miner);
+
+    Witnessed(bytes_from_signed_tx(&witnessed_block))
+}
+
+async fn get_decided_block_result(block: &models::block::Block, req_miner_address: &PublicKey, next_block: ExpectedBlock) -> Option<PostBlockResult> {
+    let next_block_miner: Miner = next_block.miner.clone();
+    let next_block_number: i32 = next_block.number.clone();
+    let burned_by_me: bool = next_block.is_miner_burned_by_me(&next_block_miner).await;
+    let burn_proof: Option<BurnProofs> = next_block.current_burn_proof();
+    let current_miner_decision: Option<&MinerBlockDecision> = next_block.decisions.get(&next_block_miner.address);
+    let req_miner_decision: Option<&MinerBlockDecision> = next_block.decisions.get(req_miner_address);
+
+
+    return if next_block_miner.address != req_miner_address.clone() || next_block_number != block.number {
+        if block.number < next_block_number {
+            // TODO: Send witness / rejection
+            Some(Witnessed(Bytes("derp".as_bytes().to_vec())))
+        } else if block.number > next_block_number {
+            Some(NotConsidered())
+        } else {
+            Some(get_res_from_decision(req_miner_decision.clone()))
+        }
+    } else {
+        match get_res_from_decision(current_miner_decision) {
+            NotConsidered() => match burned_by_me {
+                true => {
+                    let bytes = bytes_from_signed_txs(burn_proof.unwrap().values());
+                    Some(Rejected(bytes))
+                },
+                false => Some(NotConsidered())
+            }
+            x => Some(x)
+        }
+    }
+}
+
+fn get_res_from_decision(decision: Option<&MinerBlockDecision>) -> PostBlockResult {
+    return match decision {
+        Some(Burned(burn_txs)) => Rejected(bytes_from_signed_txs(burn_txs.values())),
+        Some(Accepted(witnessed_block)) => Witnessed(bytes_from_signed_tx(witnessed_block)),
+        None => NotConsidered()
+    }
+}
+
+fn bytes_from_signed_txs<'a, I>(signed_txs: I) -> Vec<Bytes>
+    where
+        I: Iterator<Item = &'a Sign1>,
+{
+    signed_txs
+        .map(|t| bytes_from_signed_tx(t))
+        .collect()
+}
+
+fn bytes_from_signed_tx(signed_tx: &Sign1) -> Bytes {
+    Bytes::from(serde_cbor::to_vec(signed_tx).unwrap())
```

Comment:
One thing we'll eventually need to take into account: `unwrap` could cause a panic if we're given bad data. Since we have a trusted miner set we're good for now but something to keep in mind in the future,